include: net: Link error when inet_pton() is used and wifi offloading is enabled

**Describe the bug**
When wifi offloading is enabled on the cc3220sf_launchxl, a call to inet_pton() would cause a link error similar to the following:

```
    ../app/libapp.a(main.c.obj): In function `zsock_inet_pton':
    /home/zephyr-work/samples/net/mqtt_publisher/build/zephyr/include/generated/syscalls/socket.h:33: undefined reference to `_impl_zsock_inet_pton'
```


**To Reproduce**
Steps to reproduce the behavior:
1. add a call to inet_pton in main()
2. mkdir build; cd build
2. cmake .. -DBOARD=cc3220xl_launchxl
3. make
4. See error

**Expected behavior**
I expect the code to build.

**Impact**
This prevents applications from using inet_pton() when wifi offloading is enabled.

**Environment (please complete the following information):**
 - OS: Ubuntu 18.04
 - Toolchain: Zephyr SDK  0.9.5
 - Commit SHA where this was reproduced: b3bb689cab6826e7a9162c2d4dfc3b3687adc0ae

--- a/include/net/socket.h
+++ b/include/net/socket.h
@@ -261,6 +261,11 @@ static inline void freeaddrinfo(struct zsock_addrinfo *ai)
 
 #define addrinfo zsock_addrinfo
 
+static inline int inet_pton(sa_family_t family, const char *src, void *dst)
+{
+	return zsock_inet_pton(family, src, dst);
+}
+
 #else
 
 struct addrinfo {
@@ -275,6 +280,21 @@ struct addrinfo {
 };
 
 #include <net/socket_offload.h>
+
+static inline int inet_pton(sa_family_t family, const char *src, void *dst)
+{
+	if ((family != AF_INET) && (family != AF_INET6)) {
+		errno = EAFNOSUPPORT;
+		return -1;
+	}
+
+	if (net_addr_pton(family, src, dst) == 0) {
+		return 1;
+	} else {
+		return 0;
+	}
+}
+
 #endif /* !defined(CONFIG_NET_SOCKETS_OFFLOAD) */
 
 #define POLLIN ZSOCK_POLLIN
@@ -290,11 +310,6 @@ static inline char *inet_ntop(sa_family_t family, const void *src, char *dst,
 			      size_t size)
 {
 	return net_addr_ntop(family, src, dst, size);
-}
-
-static inline int inet_pton(sa_family_t family, const char *src, void *dst)
-{
-	return zsock_inet_pton(family, src, dst);
 }
 
 #define EAI_BADFLAGS DNS_EAI_BADFLAGS

